moveit_core/collision_detection_fcl/src/collision_env_fcl.cpp

```diff
@@ -166,7 +166,8 @@ void CollisionEnvFCL::constructFCLObjectWorld(const World::Object* obj, FCLObjec
     FCLGeometryConstPtr g = createCollisionGeometry(obj->shapes_[i], obj);
     if (g)
     {
-      auto co = new fcl::CollisionObjectd(g->collision_geometry_, transform2fcl(obj->shape_poses_[i]));
+      auto co = new fcl::CollisionObjectd(g->collision_geometry_,
+                                          transform2fcl(getWorld()->getGlobalShapeTransform(obj->id_, i)));
```

Comment:
Same here:
```suggestion
      auto co = new fcl::CollisionObjectd(g->collision_geometry_,
                                          transform2fcl(obj->global_shape_poses_[i]));
```
Do we need `getGlobalShapeTransform(s)` at all?